tox.ini

```diff
@@ -25,98 +55,75 @@ deps =
     sqlalchemy
     Flask-SQLAlchemy
     future
-    django22: Django==2.2.*
-    django30: Django==3.0.*
-    django31: Django==3.1.*
-    django32: Django==3.2.*
-    django{22,30,31,32}: django-fake-model
+    django-2: Django>=2.0,<3.0
+    django-3: Django>=3.0,<4.0
+    django-{2,3}: django-fake-model
     pynamodb >= 3.3.1
     psycopg2
-    pg8000
+    pg8000 <=1.20.0
```

Comment:
This is probably best done as having the first part of the `deps =` section as dependencies we know we need for every test, then those for python specific versions, then for `core`, then for `ext`:

```
deps =
    ; Testing packages
    pytest > 3.0.0
    pytest-benchmark
    coverage==4.5.4
    codecov

    ; Deps for all test environments
    future
    wrapt

    ; Python 2.7 only deps

    py{27}: enum34
    py{27}: mock

    ; Python 3.4 only deps

    py34: typing >= 3.7.4.3

    ext-requests: requests

    ext-bottle: bottle >= 0.10 webtest

    ext-flask: flask >= 0.10
    ; TODO: Is this the correct test to download this for?
    ext-flask: Flask-SQLAlchemy

    ; TODO: Will this match `sqlalchemy_core` too?
    ext-sqlalchemy: sqlalchemy
    ; TODO: Is this the correct test to download this for?
    ext-sqlalchemy: Flask-SQLAlchemy

    ext-django2: Django>=2.0,<3.0

    ext-django3: Django>=3.0,<4.0

    ext-django: django-fake-model

    ext-pynamodb >= 3.3.1

    ext-psycopg2: psycopg2 testing.postgresql

    ext-pg8000: pg8000 <=1.20.0

    ext-pymysql: testing.mysqld
    py{27,34,35}-ext-pymysql: pymysql < 1.0.0
    py{36,37,38,39}-ext-pymysql: pymysql >= 1.0.0

    ; Python 3.5+ only deps

    py{35,36,37,38,39}-ext-aiobotocore: aiobotocore >= 0.10.0

    py{35,36,37,38,39}-ext-aiohttp: aiohttp >= 3.0.0
    py{35,36,37,38,39}-ext-aiohttp: pytest-aiohttp
```

Reply:
I think by doing this you might find some packages need to have their packages brought in explicitly because they are not brought in as direct dependencies anymore.

For example `ext-botocore` may have been getting `botocore` because we downloaded `aiobotocore` on every test, but if we split them like this then we would have to do:

```
ext-botocore: botocore
```

to pull it in specifically.